src/Tribe/Views/V2/iCalendar/Links/Link_Abstract.php

```diff
@@ -153,6 +155,41 @@ public function filter_tec_views_v2_single_subscribe_links( $links ) {
 		return $links;
 	}
 
+	/**
+	 * Hide the subscribe link on user agents with selected keywords.
+	 *
+	 * @since TBD
+	 *
+	 * @param bool $visible Whether the subscribe link should be visible.
+	 *
+	 * @return bool
+	 */
+	public function maybe_hide_subscribe_link( $visible ): bool {
+		// Bail if we're on the single event page.
+		// Note, subscription on single event pages works.
+		if ( is_single() ) {
+			return $visible;
+		}
+
+		// Bail if the user agent is not set.
+		if ( ! isset( $_SERVER['HTTP_USER_AGENT'] ) ) {
+			return $visible;
+		}
+
+		$current_user_agent = strtolower( $_SERVER['HTTP_USER_AGENT'] ); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPressVIPMinimum.Variables.RestrictedVariables.cache_constraints___SERVER__HTTP_USER_AGENT__
```

Comment:
sanitize_text_field and don't ignore `InputNotSanitized`

Reply:
Could also combine this with the above check:
```
		$current_user_agent = sanitize_text_field( $_SERVER['HTTP_USER_AGENT'] ?? '' );
		
		// Bail if the user agent is empty.
		if ( empty( $current_user_agent ) ) {
			return $visible;
		}
```

Also - you use `stripos` (case insensitive) below - why bother with `strtolower` here? Redundant.